### ISSUE TITLE
InfluxDB Error: Internal Server Error: interface conversion: interface {} is  nil, not string

### ISSUE THREAD

**Issue by pprkut** [grafana/grafana#63936]
**What happened**:

We update our grafana instance from 9.3.6 to 9.4.1 and now some of our dashboards show an error:

```
InfluxDB Error: Internal Server Error - Check the Grafana server logs for the detailed error message.
```

Checking the logs I see this message:

```
interface conversion: interface {} is  nil, not string
```
with this stacktrace:
```
/usr/local/go/src/runtime/iface.go:262 (0x414e29)
/drone/src/pkg/tsdb/influxdb/response_parser.go:103 (0x34713e4)
/drone/src/pkg/tsdb/influxdb/response_parser.go:40 (0x346f3be)
/drone/src/pkg/tsdb/influxdb/influxdb.go:145 (0x3469548)
/drone/src/pkg/plugins/backendplugin/coreplugin/core_plugin.go:85 (0x3511090)
/drone/src/pkg/plugins/plugins.go:285 (0x12114f4)
/drone/src/pkg/plugins/manager/client/client.go:48 (0x3535e11)
/drone/src/pkg/plugins/backendplugin/instrumentation/instrumentation.go:42 (0x353481e)
/drone/src/pkg/plugins/backendplugin/instrumentation/instrumentation.go:103 (0x35357b5)
/drone/src/pkg/plugins/manager/client/client.go:44 (0x3535790)
/drone/src/pkg/services/pluginsintegration/clientmiddleware/httpclient_middleware.go:71 (0x353ad3e)
/drone/src/pkg/services/pluginsintegration/clientmiddleware/cookies_middleware.go:90 (0x3539f7b)
/drone/src/pkg/services/pluginsintegration/clientmiddleware/oauthtoken_middleware.go:100 (0x353b8fc)
/drone/src/pkg/services/pluginsintegration/clientmiddleware/clear_auth_headers_middleware.go:48 (0x353950a)
/drone/src/pkg/services/pluginsintegration/clientmiddleware/tracing_header_middleware.go:53 (0x353c0ca)
/drone/src/pkg/plugins/manager/client/decorator.go:35 (0x35372a8)
/drone/src/pkg/services/query/query.go:236 (0x1f8c8c1)
/drone/src/pkg/services/query/query.go:84 (0x1f8ac51)
/drone/src/pkg/api/metrics.go:55 (0x2da690b)
/drone/src/pkg/api/routing/routing.go:17 (0x190c8a1)
/drone/src/pkg/api/response/web_hack.go:40 (0x11da175)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:134 (0x11ced54)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/services/contexthandler/contexthandler.go:221 (0x18f9db9)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/render.go:44 (0x11cf915)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/macaron.go:140 (0x11cedfb)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/middleware/csrf/csrf.go:64 (0x20a8625)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/middleware/recovery.go:171 (0x1907221)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/middleware/logger.go:42 (0x1904981)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/middleware/request_metrics.go:59 (0x1908049)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/middleware/request_tracing.go:88 (0x190979a)
/usr/local/go/src/net/http/server.go:2109 (0x7ae40e)
/drone/src/pkg/web/context.go:49 (0x11cd6bd)
/drone/src/pkg/web/router.go:155 (0x11d0e8d)
/drone/src/pkg/web/router.go:211 (0x11d1af5)
/drone/src/pkg/web/macaron.go:166 (0x11cf270)
/usr/local/go/src/net/http/server.go:2947 (0x7b180b)
/usr/local/go/src/net/http/server.go:1991 (0x7acee6)
/usr/local/go/src/runtime/asm_amd64.s:1594 (0x4783c0)
```

**What you expected to happen**:

Dashboard should load as expected.

**How to reproduce it (as minimally and precisely as possible)**:

It seems to be happening with a very specific dashboard, but I haven't figured out yet what the deciding factor is :-/
I'll test a bit more and report back when I have more findings.

The dashboard doesn't have much though. It's just one table panel essentially.

**Environment**:
- Grafana version: 9.4.1
- Data source type & version: InfluxDB 5.0.0
- OS Grafana is installed on: CentOS 7
- User OS & Browser: Linux / Firefox 
- Grafana plugins: -
- Others: -

**Comment by pprkut** [comment]
Temporarily downgraded one instance to 9.3.8, and there it's working again. I can leave a different one running at 9.4.1 though to debug further (if needed)

**Comment by usommerl** [comment]
I ran into the same issue today. I'm using InfluxDB 1.8.2. My dashboards are working fine in Grafana 9.3.8. The error occurs when I use Grafana 9.4.1 or 9.4.2. I was able to trigger the described error via the Explore UI with this query: 

`SELECT * FROM "measurement_name" WHERE $timeFilter`

The query fails when I set `$timeFilter` via the time picker to `24h`  or to a larger value. I'm not sure whether this has something to do with the data in my database or how Grafana expands the `$timeFilter` variable. I also noticed that in 9.4.x the Grafana query inspector does not show the expanded value for the `$timeFilter` variable anymore.

**Comment by pasce1998** [comment]
I had the same issue today with InfluxQL queries. Queries without `$timeFilter` worked just as expected but the ones with didn't. Downgrading back to 9.3.8 worked but is not a permanent solution.

**Comment by melroy89** [comment]
I also suffer from the same issue.

**Comment by guenter-ms** [comment]
Same Problem when migrated from 9.2.4 to 9.4.3. After downgrade OK.

**Comment by hacki11** [comment]
Same for me after updating to 9.4.3.
I have a stat panel which displays the last string entry of a given time frame from an InfluxDB v2.6.1 measurement.

**Comment by melroy89** [comment]
@tolzhabayev could someone of Grafana can take a serious look at this regression issue? Please.

**Comment by nickidw** [comment]
Have the same issue after upgrading from 9.3.6 to 9.4.3, also using influxdb.

**Comment by itsmylife** [comment]
I think this was resolved here https://github.com/grafana/grafana/pull/64556
@bohandley Could you please verify this?

**Comment by bohandley** [comment]
@itsmylife yes this was issue was fixed with #64556 and will be released in v9.4.5

**Comment by nickidw** [comment]
Confirmed fixed yesterday with docker grafana/grafana-oss:main which showed version v9.5.0-105767pre

**Comment by lux4rd0** [comment]
Using grafana/grafana-oss:main I don't get the errors anymore - but my tables are a little off with a new lowercase "time" coming back from my query instead of "Time" being returned. Because I use "Join By Field" transformations, things look broken until I go back and clean up all of my tables.

**Comment by melroy89** [comment]
I'm still waiting for a new Jammy release https://repos.influxdata.com/ubuntu/dists/jammy/

**Comment by bohandley** [comment]
@lux4rd0 @melroy89 As part of a more robust fix and to handle the issue with the lower case "time" we have reverted a change that effects both the interface {} error and the "Time" to "time" issue.

https://github.com/grafana/grafana/pull/64842

This fix will be released in v9.4.6
